Base/Python/slicer/util.py

```diff
@@ -529,20 +557,42 @@ def setToolbarsVisible(visible, ignore=None):
     pass
 
 def setMenuBarsVisible(visible, ignore=None):
-  """Show/hide all menu bars, except those listed in
-  ignore list."""
-  for menubar in mainWindow().findChildren('QMenuBar'):
+  """Show/hide all menu bars, except those listed in ignore list.
+
+  If there is no main window then the function has no effect.
+  """
+  mw = mainWindow()
+  if not mw:
+    return
+  for menubar in mw.findChildren('QMenuBar'):
     if ignore is not None and menubar in ignore:
       continue
     menubar.setVisible(visible)
 
 def setPythonConsoleVisible(visible):
-  """Show/hide Python console."""
-  mainWindow().pythonConsole().parent().setVisible(visible)
+  """Show/hide Python console.
+
+  If there is no main window then the function has no effect.
+  """
+  mw = mainWindow()
+  if not mw:
+    return
+  mw.pythonConsole().parent().setVisible(visible)
 
 def setStatusBarVisible(visible):
-  """Show/hide status bar"""
-  mainWindow(verbose=False).statusBar().setVisible(visible)
+  """Show/hide status bar
+
+  If there is no main window or status bar then the function has no effect.
+  """
+  mw = mainWindow()
+  if not mw:
+    return
+  try:
+    statusBar = mw.statusBar()
```

Comment:
this try/except is not needed. From the Qt documentation:

```
Returns the status bar for the main window. This function creates and returns an empty status bar if the status bar does not exist.
```

Reply:
If we keep the `try/except`, I suggest we use a specific exception

Reply:
fixed